blog/_src/posts/2017-02-05-on-application.md

```diff
@@ -0,0 +1,108 @@
+    Title: On Application
+    Date: 2017-02-05T15:17:41
+    Tags:
+
+*posted by Jack Firth*
+
+Today I wanted to write about function application. Specifically, how to redefine and customize
+application with Racket. We'll also look at some Racket packages the define interesting and useful
+forms of function application.
+
+## The theory
+
+Application defines one half of *lambda calculus*, the formal model underlying much of modern
+functional programming. The other half is *abstraction*, which creates new functions. Creating and
+applying functions lies at the heart of Racket and many other functional languages.
+
+So how are functions applied in Racket? What makes `(if (< 5 10) 'foo 'bar)` a macro use and
+`(< 5 10)` a function use?
+
+That's actually a trick question, because function application *is* a macro in Racket. During macro
+expansion, the Racket expander inspects the first element of an expression to determine if it has a
```

Comment:
add link to "Expansion Steps" in docs
http://docs.racket-lang.org/reference/syntax-model.html#%28part._expand-steps%29